sympy/series/fourier.py

```diff
@@ -440,16 +438,127 @@ def __sub__(self, other):
         return self.__add__(-other)
 
 
-class FiniteFourierSeries(Basic):
+class FiniteFourierSeries(FourierSeries):
+
     def __new__(cls, *args):
-        obj = Basic.__new__(cls, *args)
-        return obj
+        if type(args[2]) == tuple and len(args[2]) == 3:
```

Comment:
Can you add a docstring explaining the parameters. In which case should `args[2]` be a tuple? Could it be more generally any sequence? Would it be useful to have `f`, `limits` and `exprs` instead of `*args`?

Reply:
> Can you add a docstring explaining the parameters. In which case should `args[2]`

I will add docstrings.

> In which case should `args[2]` be a tuple? Could it be more generally any sequence?

Since the new class is very similar to `FourierSeries` class I tried to reuse most of the code. In `shift` method no change is applied to `(a0, an, bn)` and hence I made the `__new__` method to accept such conditions. In other cases `arg[2]` is a simplified form of expression.

> Would it be useful to have `f`, `limits` and `exprs` instead of `*args`?

Since `args` is a tuple I needed to assign values. This way it is more readable.

Reply:
I think that `__new__(cls, f, limits, exprs)` could also be readable. It would also automatically check that there are exactly three parameters in addition to `cls` so, for example, `args[2]` would not raise an exception if there are less that three parameters.

Reply:
I tried to follow the code of `FourierSeries` class to mimic it and keep the code consistent. I can implement it this way but I think the user would mostly interact with `fourier_series` function where we already control the number of parameters.

Reply:
I also believe that this would most often be called via `fourier_series`. But I think that having explicit parameter names could still be useful. It would probably make the code easier to maintain.

Reply:
I also think it would benefit further development of the code.
Most of the docstring for `FiniteFourierSeries ` class follows that of `FourierSeries` class. And the code inside __new__ function is local to the function. Most of work on this issue is to change the form of expression to match the form of Fourier series. The code inside __new__ is just doing that. I don't think I should write a docstring for that part but I can write comments explaining them. Please suggest.

Reply:
Perhaps you could document the class as recommended in [numpydocs](https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes).